src/lib/loader.js

```diff
@@ -38,26 +40,33 @@ async function loadEntrypoint(url) {
 }
 
 /**
- * Gets the partial content of the target normalized URL. Returns null if aborted.
+ * Gets the HTML content of the target normalized URL. Returns null if aborted.
  *
- * If the partial is missing (i.e., 404) this throws an error. This means that
+ * If the HTML is missing (i.e., 404) this throws an error. This means that
  * requests to missing pages will do an additional network round-trip. This is
  * important as there might be a configured redirect.
  *
  * @param {string} url of the page to fetch.
  * @param {!AbortSignal=} signal
- * @return {?{raw: string, title: string, offline: (boolean|undefined)}}
+ * @return {{offline: boolean, html: string}}
  */
-export async function getPartial(url, signal) {
-  if (!url.endsWith('/')) {
-    throw new Error(`partial unsupported for non-folder: ${url}`);
+export async function getHTML(url, signal) {
+  if (!(url.endsWith('/') || url.endsWith('.html'))) {
+    throw new Error(`can't fetch HTML for unsupported URL: ${url}`);
```

Comment:
This error message seems like it might be confusing. Like, if I fetch foo.html and it tells me it can't fetch unsupported "HTML" for foo.html.

Is the issue that we have to cache the url in the service worker and it needs to specifically be a directory that ends in a trailing slash? I'm not sure I quite grok what's going on

Reply:
Maybe i'm reading it wrong, but it seems like it wouldn't allow files that end in .html because it checks `url.endsWith('.html')` and considers that a reason to throw. Am I missing something?

Reply:
!(a || b) is the same as !a && !b.

Or to put it differently...

```js
const valid = url.endsWiith('/') || url.endsWith('.html');
if (!valid) {
  throw
}
```

Reply:
ah I read it as: `((!url.endswith('/') || url.endsWith('.html'))`